test/controllers/update/company.profile.controller.spec.ts

```diff
@@ -0,0 +1,106 @@
+
+describe("Confirm company data", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("Get company data by Id", () => {
+
+    test("Redirection to What is your OE number page if OE number is undefined", async () => {
+      mockGetApplicationData.mockReturnValueOnce({});
+      const resp = await request(app).get(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL);
+      expect(resp.statusCode).toEqual(302);
+      expect(resp.redirect).toEqual(true);
+      expect(resp.text).toContain(config.OVERSEAS_ENTITY_QUERY_URL);
+    });
+
+    test("OE error key is set when OE data does not exist", async () => {
+      mockGetApplicationData.mockReturnValueOnce({});
+      const resp = await request(app).get(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL);
+      expect(resp.statusCode).toEqual(302);
+      expect(mockSetExtraData).toBeCalledWith(req.session, {
+        [OeErrorKey]: "The Overseas Entity with OE number  is not valid or does not exist."
+      });
+      expect(mockSetExtraData).toHaveBeenCalledTimes(1);
+      expect(resp.header.location).toEqual(config.OVERSEAS_ENTITY_QUERY_URL);
+    });
+
+    test("confirm and continue button is rendered", async () => {
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+      mockGetOeCompanyDetails.mockReturnValue( { oe_number: '12345678' } );
+
+      const resp = await request(app).get(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL);
+      expect(resp.statusCode).toEqual(200);
+      expect(resp.text).toContain(CONFIRM_AND_CONTINUE_BUTTON_TEXT);
+    });
+
+    test("Change company link is rendered", async () => {
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+      mockGetOeCompanyDetails.mockReturnValue( APPLICATION_DATA_MOCK );
+      const resp = await request(app).get(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL);
+      expect(resp.statusCode).toEqual(200);
+      expect(resp.text).toContain(CHANGE_COMPANY_TEST);
+    });
+
+    test(`redirect to the ${config.UPDATE_OVERSEAS_ENTITY_DETAILS_PAGE} page`, async () => {
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+      mockGetOeCompanyDetails.mockReturnValue( APPLICATION_DATA_MOCK );
+      const resp = await request(app).post(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL);
+      expect(resp.statusCode).toEqual(302);
+      expect(resp.redirect).toEqual(true);
+    });
+
+    test(`OE number is retrieved from session data`, async () => {
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+      mockGetOeCompanyDetails.mockReturnValue( APPLICATION_DATA_MOCK );
+      const resp = await request(app).get(config.CONFIRM_OVERSEAS_ENTITY_PROFILES_URL);
+      expect(resp.statusCode).toEqual(200);
+      expect(mockGetOeCompanyDetails).toHaveBeenCalled();
+    });
+
+    test('catch error when rendering the page', async () => {
+      mockLoggerDebugRequest.mockImplementationOnce( () => { throw new Error(ANY_MESSAGE_ERROR); });
```

Comment:
this is very fragile - if someone removes the logger.debugRequest the test will break

Reply:
Is there a better alternative? If mockLoggerDebugRequest is deleted, all CH test will fail because it is the approach in use

Reply:
ok. I'd have made the render throw the error but as other CH tests take this approach.